### PR TITLE
[Merged by Bors] - feat: `a / c ≡ b / c [PMOD p]`

### PR DESCRIPTION
**Pull request by YaelDillies**
Also fix a few misported names.

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/ModEq.lean [hunk 1]
```diff
@@ -321,6 +321,10 @@ theorem int_cast_modEq_int_cast {a b z : ℤ} : a ≡ b [PMOD (z : α)] ↔ a 
   norm_cast
 #align add_comm_group.int_cast_modeq_int_cast AddCommGroup.int_cast_modEq_int_cast
 
+@[simp, norm_cast]
+lemma intCast_modEq_intCast' {a b : ℤ} {n : ℕ} : a ≡ b [PMOD (n : α)] ↔ a ≡ b [PMOD (n : ℤ)] := by
```
Comment (urkud) [inline Mathlib/Algebra/ModEq.lean:325]: Why `intCast`, not `int_cast`? The previous lemma uses `int_cast`.

Comment (YaelDillies) [reply Mathlib/Algebra/ModEq.lean:325]: It's one operation, so should be one atom. Let me fix the previous lemma.

Comment (urkud) [reply Mathlib/Algebra/ModEq.lean:325]: OTOH, we have [Function.Involutive.list_map](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/List/Basic.html#Function.Involutive.list_map)

Comment (YaelDillies) [reply Mathlib/Algebra/ModEq.lean:325]: I don't think we should block this PR on a naming decision that mathlib isn't consistent about.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**